flocker/node/agents/cinder.py

```diff
@@ -902,8 +946,30 @@ def cinder_from_configuration(region, cluster_id, **config):
         _nova_servers=nova_client.servers
     )
     return CinderBlockDeviceAPI(
-        cinder_volume_manager=logging_cinder,
+        cinder_volume_manager=wrapped_cinder_volume_manager,
         nova_volume_manager=logging_nova_volume_manager,
         nova_server_manager=logging_nova_server_manager,
         cluster_id=cluster_id,
     )
+
+
+class Cinder1to2Adapter(proxyForInterface(ICinderVolumeManager, "_client_v2")):
+    """
+    Deal with annoying differences in the method signatures between
+    cinderclient.client.{v1,v2}.volumes.VolumeManager
+    """
+    def create(self, size, metadata=None, display_name=None):
+        """
+        v1 uses display_name rather than name.
+        """
+        return self._client_v2.create(
+            size=size,
+            metadata=metadata,
+            name=display_name
+        )
+
+
+CINDER_V1_ADAPTERS = {
```

Comment:
I'd prefer unifying this data structure with the supported versions data structure, as they should never be out of sync.